kse/src/org/kse/gui/crypto/JValidityPeriod.java

```diff
@@ -163,9 +163,14 @@ private void typeChanged(int periodType) {
 			break;
 		}
 
-		if (((Comparable<Object>) model.getMaximum()).compareTo(value) >= 0
-				&& ((Comparable<Object>) model.getMinimum()).compareTo(value) <= 0) {
-			model.setValue(value);
+		try {
+			if (((Comparable<Object>) model.getMaximum()).compareTo(value) >= 0
+					&& ((Comparable<Object>) model.getMinimum()).compareTo(value) <= 0) {
+				model.setValue(value);
+			}			
+		}
+		catch(Exception e) {
+			//ignore
```

Comment:
I have not managed to solve that problem, it happens when I change validity period.

java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
	at java.base/java.lang.Integer.compareTo(Integer.java:59)
	at org.kse.gui.crypto.JValidityPeriod.typeChanged(JValidityPeriod.java:167)
	at org.kse.gui.crypto.JValidityPeriod.lambda$0(JValidityPeriod.java:115)

Reply:
That's because you are passing a long value in line 222 of DSignCrl 
jvpValidityPeriod.setValue(Long.valueOf(diff));
Integer is enough for days, so I would change setValue(Number) to setValue(int)

Reply:
thank you, the error no longer appears.